pkg/controller/controller.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+)
+
+func NewControllerServer(identity, nodeID string) (*ControllerServer, error) {
+	return &ControllerServer{
+		NodeID:   nodeID,
+		Identity: identity,
+	}, nil
+}
+
+type ControllerServer struct {
+	NodeID   string
+	Identity string
+}
+
+func (c *ControllerServer) ControllerGetCapabilities(ctx context.Context, req *csi.ControllerGetCapabilitiesRequest) (*csi.ControllerGetCapabilitiesResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "unimplemented")
```

Comment:
Instead of all of these, you could make `type ControllerServer` 'inherit' from `csi.UnimplementedControllerServer`.

Reply:
I'll look into this for the CSI driver! Thank for sharing